scripts/o3de/o3de/gem_properties.py

```diff
@@ -85,24 +88,17 @@ def edit_gem_props(gem_path: pathlib.Path = None,
                          f' characters, and start with a letter.  {new_name}')
             return 1
         update_key_dict['gem_name'] = new_name
-    if new_display:
-        update_key_dict['display_name'] = new_display
-    if new_origin:
-        update_key_dict['origin'] = new_origin
-    if new_type:
-        update_key_dict['type'] = new_type
-    if new_summary:
-        update_key_dict['summary'] = new_summary
-    if new_icon:
-        update_key_dict['icon_path'] = new_icon
-    if new_requirements:
-        update_key_dict['requirements'] = new_requirements
-    if new_documentation_url:
-        update_key_dict['documentation_url'] = new_documentation_url
-    if new_license:
-        update_key_dict['license'] = new_license
-    if new_license_url:
-        update_key_dict['license_url'] = new_license_url
+    
+    update_key_dict['display_name'] = new_display if new_display else ""
+    update_key_dict['origin'] = new_origin if new_origin else ""
+    update_key_dict['type'] = new_type if new_type else ""
+    update_key_dict['summary'] = new_summary if new_summary else ""
+    update_key_dict['icon_path'] = new_icon if new_icon else ""
+    update_key_dict['requirements'] = new_requirements if new_requirements else ""
+    update_key_dict['documentation_url'] = new_documentation_url if new_documentation_url else ""
+    update_key_dict['license'] = new_license if new_license else ""
+    update_key_dict['license_url'] = new_license_url if new_license_url else ""
+    update_key_dict['repo_uri'] = new_repo_uri if new_repo_uri else ""
```

Comment:
As discussed - this logic would update every value that wasn't passed into this function with an empty string so we just want the `if` block not the `else`
e.g.
```
if new_display not None:
        update_key_dict['display_name'] = new_display
```
That or an `isinstance` check.
That would let a user set the value to an empty string.